frontend/src/stores/applications.js

```diff
@@ -101,9 +45,91 @@ export const useApplicationsStore = defineStore('applications', {
       }
     },
 
+    /*
+      Facility Details page
+    */
+    checkFacilityDetailsComplete() {
+      return this.currentApplication?.primaryContactId && this.currentApplication?.expenseAuthorityId && this.currentApplication?.fiscalYearEndDate
+    },
+
+    /* 
+      Service Delivery page
+     */
+    checkServiceDeliveryComplete() {
+      return (
+        this.currentApplication?.licenceDeclaration &&
+        !isEmpty(this.currentApplication?.licences) &&
+        this.isLicenceDetailComplete() &&
+        this.isLicenceDocumentUploaded() &&
+        this.isHealthAuthorityReportUploaded()
+      )
+    },
+
+    isLicenceDetailComplete() {
+      return this.isCCOFMissingDetailComplete() && this.isSplitClassroomComplete()
+    },
+
+    isCCOFMissingDetailComplete() {
+      return this.currentApplication?.licences.every((licence) => {
+        return licence.licenceDetails?.every(
+          (detail) =>
+            LicenceService.isOperationalSpacesValid(detail.operationalSpaces) &&
+            LicenceService.isEnrolledSpacesValid(detail.enrolledSpaces) &&
+            LicenceService.isWeeksInOperationValid(detail.weeksInOperation) &&
+            !isEmpty(detail.operationFromTime) &&
+            !isEmpty(detail.operationToTime) &&
+            !isEmpty(detail.weekDays),
+        )
+      })
+    },
+
+    isSplitClassroomComplete() {
+      return this.currentApplication?.licences.every((licence) => licence.licenceDetails?.every((detail) => LicenceService.isSplitClassRoomInfoValid(detail)))
+    },
+
+    isLicenceDocumentUploaded() {
+      return this.currentApplication.licences.every((licence) => {
+        const uploadedDocument = this.currentApplication?.uploadedDocuments?.filter((document) => document.documentType?.includes(licence?.licence))
+        return uploadedDocument?.length > 0
```

Comment:
We've got a mix of isEmpty and length > 0 usage in this file when working with arrays.

Reply:
it's updated